msm/src/column_env.rs

```diff
@@ -36,19 +37,28 @@ impl<'a, F: FftField> TColumnEnvironment<'a, F> for ColumnEnvironment<'a, F> {
         &self,
         col: &Self::Column,
     ) -> Option<&'a Evaluations<F, Radix2EvaluationDomain<F>>> {
-        let witness_columns_n: usize = self.witness.len();
-        let coefficients_columns_n: usize = self.coefficients.len();
-        let crate::columns::Column::X(i) = col else {
-            todo!()
-        };
-        let i = *i;
-        if i < witness_columns_n {
-            let res = &self.witness[i];
-            Some(res)
-        } else if i < witness_columns_n + coefficients_columns_n {
-            Some(&self.coefficients[i])
-        } else {
-            None
+        let witness_length: usize = self.witness.len();
+        let coefficients_length: usize = self.coefficients.len();
+        match *col {
+            // Handling the "relation columns"
+            Self::Column::X(i) => {
+                if i < witness_length {
+                    let res = &self.witness[i];
+                    Some(res)
+                } else if i < witness_length + coefficients_length {
+                    // FIXME and add a test
+                    Some(&self.coefficients[i])
```

Comment:
Not sure but, wouldn't this be `self.coefficients[i-witness_length]` then?

Reply:
Yes... It must be fixed. Even though it is obvious to change the index, I want to fix it later. it is not used atm.

Reply:
I simply renamed var and reorg the code. The logic is the same than before.

Reply:
okay, let's not forget then

Reply:
Open https://github.com/o1-labs/proof-systems/issues/1988